From 510a0d41bdc2fe0e0e0c69e5c97196b375841564 Mon Sep 17 00:00:00 2001
From: Roberto Sartori <roberto.sartori.android@gmail.com>
Date: Tue, 14 Nov 2023 14:22:45 +0100
Subject: [PATCH] HACK: set minimum render frame rate to 60fps

On Android 14, for some reason, when a 30fps content (e.g. video) is on screen
the full UI is rendered at 30fps.

Until Google fixes this, set 60fps as lower limit to basically force the UI
to be rendered at 60fps.

Change-Id: I1133ccbb22282d80309b6b8057fd20afaa4d817f
Signed-off-by: Roberto Sartori <roberto.sartori.android@gmail.com>
---
 .../com/android/server/display/mode/DisplayModeDirector.java  | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/services/core/java/com/android/server/display/mode/DisplayModeDirector.java b/services/core/java/com/android/server/display/mode/DisplayModeDirector.java
index 18895788e4ec..d66f51a02b1f 100644
--- a/services/core/java/com/android/server/display/mode/DisplayModeDirector.java
+++ b/services/core/java/com/android/server/display/mode/DisplayModeDirector.java
@@ -234,7 +234,7 @@ public class DisplayModeDirector {
         public void reset() {
             minPhysicalRefreshRate = 0f;
             maxPhysicalRefreshRate = Float.POSITIVE_INFINITY;
-            minRenderFrameRate = 0f;
+            minRenderFrameRate = 60.0f;
             maxRenderFrameRate = Float.POSITIVE_INFINITY;
             width = Vote.INVALID_SIZE;
             height = Vote.INVALID_SIZE;
@@ -1207,7 +1207,7 @@ public class DisplayModeDirector {
         private void updateRefreshRateSettingLocked() {
             final ContentResolver cr = mContext.getContentResolver();
             float minRefreshRate = Settings.System.getFloatForUser(cr,
-                    Settings.System.MIN_REFRESH_RATE, 0f, cr.getUserId());
+                    Settings.System.MIN_REFRESH_RATE, 60.0f, cr.getUserId());
             float peakRefreshRate = Settings.System.getFloatForUser(cr,
                     Settings.System.PEAK_REFRESH_RATE, mDefaultPeakRefreshRate, cr.getUserId());
             updateRefreshRateSettingLocked(minRefreshRate, peakRefreshRate, mDefaultRefreshRate);
-- 
2.34.1

